http-protocol-binding.md

```diff
@@ -213,16 +213,75 @@ and per [RFC7230, section 3][rfc7230-section-3], HTTP headers MUST only use
 printable characters from the US-ASCII character set, and are terminated by a
 CRLF sequence with OPTIONAL whitespace around the header value.
 
-String values MUST be percent-encoded as described in [RFC3986, section
-2.4][rfc3986-section-2-4] before applying the header encoding rules described in
-[RFC7230, section 3.2.6][rfc7230-section-3-2-6].
-
-When decoding an HTTP message into a CloudEvent, these rules MUST be applied in
-reverse -- [RFC7230, section 3.2.6][rfc7230-section-3-2-6] decoding to an ASCII
-string, and then a **single round** of percent-decoding as described in
-[RFC3986, section 2.4][rfc3986-section-2-4] to produce a valid UTF-8 String.
-(Note that applying percent-decoding an incorrect number of times can result in
-message corruption or security issues.)
+When encoding a CloudEvent as an HTTP message, string values
+represented as HTTP header values MUST be percent-encoded as
+described below. This is compatible with [RFC3986, section
+2.1][rfc3986-section-2-1] but is more specific about what needs
+encoding. The resulting string does not then need any further
+quoting, due to space and quote characters being encoded.
+
+When decoding an HTTP message into a CloudEvent, any quoted string
+as described in [RFC7230, section 3.2.6][rfc7230-section-3-2-6]
+appearing as an HTTP header value MUST be decoded to an ASCII
+string, and then a **single round** of percent-decoding MUST be
+performed as described below. HTTP headers for CloudEvent attribute
+values do not support parenthetical comments, so the initial
```

Comment:
"parenthetical comments" ?

Reply:
Yes, apparently some headers can include comments, which appear in parentheses. For example, here's a user agent header value:

```text
Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:47.0) Gecko/20100101 Firefox/47.0
```

In that, the `(Windows NT 6.1; Win64; x64; rv:47.0)` part is a comment. Only a few HTTP header fields are documented to support comments, and I suggest we never include CloudEvent headers in that set :)

Reply:
A detail of HTTP that I have not paid much attention to, indeed. I think that is a semantical issue for some HTTP field values rather than an encoding question, no?

Reply:
It's part of RFC7230 section 3.2.6... so if we're telling SDK implementers they need to implement decoding as per that section of the RFC, it really helps if we're clear about what that means - in other words, that the comment part doesn't apply.